### PR TITLE
remove mount accessor from MFA config

### PR DESCRIPTION
**Pull request by hghaf099**

### CODE REVIEW

**Comment by raskchanky** [comment]
This feels like too risky of a change for this point in the release cycle. I think it's worth making a Jira to keep track of this, but I'm not sure removing this now is a good idea.

**Comment by hghaf099** [comment]
Yeah, it might be risky. Though, for entity templating, I have used existing code here
https://github.com/hashicorp/vault/blob/191868d2b11d912115b435e54539631d7d68d23e/vault/login_mfa.go#L1534-L1549

Maybe we can manage the risk? I agree that it is really last minute, and the requirements for templating is not that clear though.

File: vault/login_mfa.go [hunk 1]
```diff
@@ -1402,20 +1383,19 @@ func (c *Core) validateLoginMFAInternal(ctx context.Context, methodID string, en
 		return fmt.Errorf("MFA method configuration not present")
 	}
 
-	var alias *identity.Alias
 	var finalUsername string
 	switch mConfig.Type {
 	case mfaMethodTypeDuo, mfaMethodTypeOkta, mfaMethodTypePingID:
```
Comment (ncabatoff) [inline vault/login_mfa.go:1388]: Think we want something more like https://github.com/hashicorp/vault/blob/main/vault/policy.go#L293

**Review by ncabatoff — COMMENTED** [review]
This is definitely on the right track.  I suspect there's a tiny bit more cleanup to do, e.g. in the protos file and the endpoint help text.  We should do at least a minimal bit of manual testing before merging.

File: website/content/api-docs/secret/identity/mfa/pingid.mdx [hunk 2]
```diff
@@ -21,12 +21,7 @@ This endpoint defines an MFA method of type PingID.
   will use the Name field of Aliases associated with this mount as the username in the mapping. Note that this can
   be from the current namespace or a child namespace.
 
-- `username_format` `(string)` - A format string for mapping Identity names to MFA method names. Values to substitute should be placed in `{{}}`. For example, `"{{alias.name}}@example.com"`. If blank, the Alias's Name field is used as-is. Currently-supported mappings:
-
-  - alias.name: The name returned by the mount configured via the `mount_accessor` parameter
-  - entity.name: The name configured for the Entity
-  - alias.metadata.`<key>`: The value of the Alias's metadata parameter
-  - entity.metadata.`<key>`: The value of the Entity's metadata parameter
+- `username_template` `(string)` - A template string for mapping Identity names to MFA method names. Values to substitute should be placed in `{{}}`. For example, `"{{alias.name}}@example.com"`. If blank, the Entity's Name field is used as-is.
```
Comment (ncabatoff) [inline website/content/api-docs/secret/identity/mfa/pingid.mdx:24]: I think the example needs adjusting.

**Review by ncabatoff — APPROVED** [review]

**Review by raskchanky — APPROVED** [review]

**Pull request merged**